src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -316,9 +316,44 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 self.write_scalar(offset_ptr, dest)?;
             }
             sym::ptr_guaranteed_eq | sym::ptr_guaranteed_ne => {
-                // FIXME: return `true` for at least some comparisons where we can reliably
-                // determine the result of runtime (in)equality tests at compile-time.
-                self.write_scalar(Scalar::from_bool(false), dest)?;
+                let a = self.read_immediate(args[0])?.to_scalar()?;
+                let b = self.read_immediate(args[1])?.to_scalar()?;
+
+                let geq_mode = intrinsic_name == sym::ptr_guaranteed_eq;
```

Comment:
"geq" to me looks like "greater or equal". I think "eq_mode" or so would be a better name.